src/Apps/Notifications/NotificationsApp.tsx

```diff
@@ -25,28 +26,33 @@ const NotificationsApp: React.FC<NotificationsAppProps> = ({ me }) => {
   const { showAuthDialog } = useAuthDialog()
   const { isLoggedIn } = useSystemContext()
 
-  useEffect(() => {
-    if (!isLoggedIn) {
-      showAuthDialog({
-        mode: "Login",
-        options: {
-          title: mode => {
-            const action = mode === "Login" ? "Log in" : "Sign up"
-            return `${action} to view your notifications.`
-          },
-        },
-        analytics: {
-          contextModule: ContextModule.activity,
-          intent: Intent.login,
+  const contentRef = useRef<HTMLDivElement | null>(null)
+
+  useOnce(() => {
+    if (isLoggedIn) return
+
+    showAuthDialog({
+      mode: "Login",
+      options: {
+        title: mode => {
+          const action = mode === "Login" ? "Log in" : "Sign up"
+          return `${action} to view your notifications.`
         },
-      })
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [])
+      },
+      analytics: {
+        contextModule: ContextModule.activity,
+        intent: Intent.login,
+      },
+    })
+  })
 
-  const isNotificationsPage = match.location.pathname.startsWith(
-    "/notifications"
-  )
+  const pathname = match.location.pathname
+  const isNotificationsPage = pathname.startsWith("/notifications")
+
+  // Scroll to top of content pane when route changes
+  useEffect(() => {
+    contentRef.current?.scrollTo(0, 0)
+  }, [pathname])
```

Comment:
@dariakoko this is how you resolve https://github.com/artsy/force/pull/14133 with scrolling. The CSS solution in that previous PR created other problems I ran into. This is much simpler.

Reply:
Q around the other PR, I don't think we should be fixing things from resizing to smaller breakpoints and then to larger ones, our UI is solid but that adds crazy more bug options to pop, what do you think @dzucconi  about that kind of fixes?

Reply:
I just fix things as I run into them. The previous fix of this bug created problems with new notification layouts. There's an even better way to fix this for real and eliminate a ton of CSS but I'm not going to bother refactoring the whole thing.

Reply:
I know and it makes sense, but my question was about your opinion of the other PR fix idea, if we should be hunting this kind of bugs on our UI